### PR TITLE
planner: avoid alloc for paramMarker in buildValuesListOfInsert

### LINKED ISSUE #25992: buildValuesListOfInsert is slow when many ‘?’ in values
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

use prepare to execute

```
insert into t values(?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?), (?, ?, ?).....
```

### 2. What did you expect to see? (Required)

fast and low cpu usage in tidb

### 3. What did you see instead (Required)

hig cpu usage

image

it seems rewrite for each "?" is unessary

### 4. What is your TiDB version? (Required)

v5.0.0 (previous flamegraph's code line number is also v5.0.0.)

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

Issue Number: close #25992 

Problem Summary:

the issue can be handle by using planCache

but ParamMarker ->  Constant  rewrite should by more lightweight without alloc for each `?` 

### What is changed and how it works?

What's Changed, How it Works:

shortcut expressionRewriter like `ValueExpr`

### Related changes

- Need to cherry-pick to the release branch 5.1, 5.0 (some user is PoC on those version)

### Check List 

Tests 

- Unit test(exists test case)

Side effects

- n/a

### Release note 

- avoid alloc for paramMarker in buildValuesListOfInsert

### CODE REVIEW

**Review by jackysp — APPROVED** [review]
LGTM

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -3019,6 +3019,8 @@ func (b *PlanBuilder) buildValuesListOfInsert(ctx context.Context, insert *ast.I
 					Value:   x.Datum,
 					RetType: &x.Type,
 				}
+			case *driver.ParamMarkerExpr:
+				expr, err = expression.ParamMarkerExpression(b.ctx, x)
```
Comment (tangenta) [inline planner/core/planbuilder.go:3023]: Where is the unnecessary memory allocating code?

Comment (coocood) [reply planner/core/planbuilder.go:3023]: in the default case

**Review by coocood — APPROVED** [review]
LGTM

**Pull request merged**